app/controllers/staff/reports_controller.rb

```diff
@@ -158,4 +171,28 @@ def send_csv
       end
     end
   end
+
+  def downloadable_reports_for_beis_users
+    [active_reports_with_organisations, submitted_reports_with_organisations, in_review_reports_with_organisations, awaiting_changes_reports_with_organisations].sum.sort_by { |report| report.organisation.name }
+  end
+
+  def report_activities_sorted_by_level(report)
+    Activity.includes(:organisation).projects_and_third_party_projects_for_report(report).sort_by { |a| a.level }
+  end
+
+  def reports_have_same_quarter?
+    downloadable_reports_for_beis_users.map(&:financial_quarter).uniq.length == 1
+  end
+
+  def send_all_reports_csv
+    report_sample = downloadable_reports_for_beis_users.first
+    headers = ExportActivityToCsv.new(report: report_sample).headers
+    stream_csv_download(filename: "Reports.csv", headers: headers) do |csv|
```

Comment:
This looks exactly like what I was expecting to see here -- sorry for the disruption of adding `stream_csv_download` and changing the response types from `ExportActivityToCsv` at the same time you were working on this!

Reply:
No worries!! Glad that I managed to get it right :-)